### DISCUSSION TITLE
Does PHPMailer work in a different way with PHP 5.6 to 7.4? (Amazon AWS SES SNS X-PHP Headers)

### DISCUSSION THREAD

**Discussion by UtopieStudio** [PHPMailer/PHPMailer#2901] [General]
Hello.
I'm really confused ATM. I recently upgraded PHPMailer from 6.1.7 to 6.8, and moved from PHP5.6 to 7.4.

My app mailing seems to work exactly the same, and looped through mails are being delivered in my tests with no errors. 

Funny thing is though, I now only receive 1 single Amazon AWS delivery notification from the sent batch, that one being for the first email in the batch that was triggered. 

I re-installed PHPMailer 6.1.7 to see if it changed back, but no... so am wondering, has changing to PHP7.4 got anything to do with it.

It's sort as if now, one loop through of about 20 emails, is classed as 1 email being sent.

I'm a bit puzzled by it

**Comment by Synchro** [comment]
I can't really tell what might be amiss without seeing your code, but generally, no, there should be no difference like this across PHP versions.

Separately,  I'd recommend that you move to a supported version of PHP.

**Reply by UtopieStudio** [reply]
Well, it's complicated but it's a server with a couple of old workhorse apps built originally with Dreamweaver would you believe.

Stage 1 was getting everything working with php7.4, next stage is php 8.1/8.2 having Rector help me.

Re issue:

I have a try catch, and after the loop finishes it echoes a list "Message sent to person@email.com", and I get the test emails through... so why AWS is only crediting the first email it sent in that batch, and not the rest, EVEN THOUGH, it has to be sending them!?!?!

**Reply by Synchro** [reply]
Are you adding multiple addresses to the same message or using BCC?

**Reply by UtopieStudio** [reply]
Ah, no, they're all separate emails, in a foreach loop.

Also noticed the headers have changed since moving to php7.4, they now include "X-PHP-Filename" revealing the dir structure of the server and script names, which isn't a great idea IMO. So am thinking if headers have changed, perhaps that's something to do with it all in someway.

Since entering the world of Amazon SES, phpList, SPF, DKIM and such things, the average person has absolutely no idea how complex the world of sending emails is. It's exhausting!

**Reply by Synchro** [reply]
You can turn that off in php.ini. Yes, email can be quite painful!
If you enable debug logging (set `$mail->SMTPDebug = 2`), you'll be able to look at the SMTP traffic and see exactly what is being submitted to the mail server.

**Comment by UtopieStudio** [comment]
SMTPdebug helped me find the cause, in a round-a-bout way.

I use the same settings on a couple different pages so put them in an inc file and included in the pages.
I must have changed it to include_once - for some reason - and that was the cause. 
inc file setting:

(I realise this is not logical really, I'm already using env variables. Just got carried away trying to clean up the old Dreamweaver spaghetti code!)
```
$mail->SMTPDebug = 0; 									
$mail->isSMTP();  										
$mail->Host = $_ENV['MAIL_HOST']; 						
$mail->SMTPAuth = true; 								

$mail->Username = $_ENV['MAIL_USER']; 					
$mail->Password = $_ENV['MAIL_PASS']; 					

// $mail->SMTPSecure = 'tls';    						
$mail->SMTPSecure = $_ENV['MAIL_SECURITY'];         
$mail->Port = $_ENV['MAIL_PORT'];  
```
SMTPDebug output 1 email's details, the same email as the SNS delivery receipt, but after that listed all the sent to emails after triggering $mail->send().
Setting to just 'include' meant the debug output and 'sent' echos were intertwined one after the other as it should be.

So still find it strange how all the emails were being delivered via AWS, but not issuing SNS email receipts, AND, only showing 1 email as being sent on the SES dashboard. It's like a hack to over-send on your daily quoter! o_O

Thanks so much, Synchro.

**Comment by UtopieStudio** [comment]
p.s. Quickie re X-PHP headers to help anyone else looking for the same issue.

Amazingly my shared hosting allowed me to add to my .htaccess:

```
<IfModule mod_headers.c>
  Header unset X-PHP-Script
  Header unset X-PHP-Filename
</IfModule>
```
I relied on Chat-GPT to give me this hint, Google search has become untenable for finding reliable information.
